platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/CommonReport.kt

```diff
@@ -224,14 +249,14 @@ class ConfigurationCacheReport(
     }
 
     private
-    var state: State = State.Idle { kind, problem ->
+    var state: State = State.Idle { problem ->
+
         State.Spooling(
             temporaryFileProvider,
-            reportFileName,
-            executorFactory.create("Configuration cache report writer", 1),
-            CharBuf::class.java.classLoader,
-            ::decorateProblem
-        ).onDiagnostic(kind, problem)
+            reportFileName.replace(" ", "-"),
```

Comment:
🤔  Maybe we shouldn't mix the concerns? I think the report file name was always intended to have dashes, and now it's not

Reply:
Do you mean we should have a separate parameter for filename and context string?